doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -452,23 +430,32 @@ commit`` to finish the merge process.
 If you want to understand the changes that occurred upstream, see
 `Read commit messages`_ below.
 
+.. _How to modify git history:
+
+How to modify git history
++++++++++++++++++++++++++
+
+The following commands are **unsafe** if the changes they overwrite have been
+pulled by other repositories.
```

Comment:
I'm not sure I'd use "unsafe" here - I'd just say something like "Changing git history can cause problems if the commits have been pulled to other repositories" or something along those lines.